src/main/java/frc/robot/utils/DutyCycleInterrupt.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.utils;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.SynchronousInterrupt;
+
+/**
+ * DutyCycleInterrupt
+ */
+public class DutyCycleInterrupt {
+
+    private final SynchronousInterrupt _interrupt;
+
+    private static final double maxFreqHz = 268; // From the datasheet
+    private static final double timeoutMS = 2 * (1 / maxFreqHz) / 1000; // From the datasheet
+
+    public DutyCycleInterrupt(SynchronousInterrupt interrupt) {
+        this._interrupt = interrupt;
+        this._interrupt.setInterruptEdges(true, true); // enable rising and falling edge interrupts
+
+    }
+
+    public DutyCycleInterrupt(DigitalInput source) {
+        this(new SynchronousInterrupt(source));
+    }
+
+    public double getFrequencyHz() {
+        double lastTime = _interrupt.getRisingTimestamp();
+        double time = System.currentTimeMillis();
+        double newTime;
+        do {
+            newTime = _interrupt.getRisingTimestamp();
+        } while (newTime == lastTime && (System.currentTimeMillis() - time) <= timeoutMS);
+
+        if (newTime == lastTime) {
```

Comment:
this is statment is kinda weird, because it guarantees a divison by 0 in the next line.